Python/instrumentation.c

```diff
@@ -1303,27 +1283,21 @@ initialize_tools(PyCodeObject *code)
     for (int i = 0; i < code_len; i++) {
         _Py_CODEUNIT *instr = &_PyCode_CODE(code)[i];
         int opcode = instr->op.code;
-        int oparg = instr->op.arg;
-        if (opcode == ENTER_EXECUTOR) {
-            _PyExecutorObject *exec = code->co_executors->executors[oparg];
-            opcode = exec->vm_data.opcode;
-            oparg = exec->vm_data.oparg;
-        }
-        else if (opcode == INSTRUMENTED_LINE) {
+        if (opcode == INSTRUMENTED_LINE) {
             opcode = code->_co_monitoring->lines[i].original_opcode;
         }
-        assert(opcode != ENTER_EXECUTOR);
         bool instrumented = is_instrumented(opcode);
         if (instrumented) {
             opcode = DE_INSTRUMENT[opcode];
             assert(opcode != 0);
         }
+        assert(opcode != ENTER_EXECUTOR);
```

Comment:
It was added to test https://github.com/python/cpython/issues/107265#issuecomment-1693405607

Reply:
Maybe the test could reproduce what you did in that comment?